paper-calendar.html

```diff
@@ -251,84 +249,84 @@
-        date: {
-          type: Date,
-          notify: true,
-          value: function() {
-            return new Date();
+    class PaperCalendar extends Polymer.mixinBehaviors([Polymer.IronResizableBehavior], Polymer.Element) {
+      static get is() { return 'paper-calendar'; }
+      static get properties() {
+        return {
+          /*
+           * The selected date
+           */
+          date: {
+            type: Date,
+            notify: true,
+            value: function () {
+              return new Date();
+            },
+            observer: '_dateChanged'
+          },
+          /**
+           * The current locale
+           */
+          locale: {
+            type: String,
+            value: 'en',
+            notify: true,
+            observer: '_localeChanged'
+          },
+          /**
+           * The minimum allowed date
+           */
+          minDate: {
+            type: Date,
+            value: null
+          },
+          /**
+           * The maximum allowed date
+           */
+          maxDate: {
+            type: Date,
+            value: null
+          },
+          /**
+           * The currently selected month (1-12)
+           */
+          currentMonth: {
+            type: Number
           },
-          observer: '_dateChanged'
-        },
-        /**
-         * The current locale
-         */
-        locale: {
-          type: String,
-          value: 'en',
-          notify: true,
-          observer: '_localeChanged'
-        },
-        /**
-         * The minimum allowed date
-         */
-        minDate: {
-          type: Date,
-          value: null
-        },
-        /**
-         * The maximum allowed date
-         */
-        maxDate: {
-          type: Date,
-          value: null
-        },
-        /**
-         * The currently selected month (1-12)
-         */
-        currentMonth: {
-          type: Number
-        },
-        /**
-         * The currently selected year
-         */
-        currentYear: {
-          type: Number
-        },
-        _contentClass: String,
-        _months: Array,
-        _firstDayOfWeek: Number,
-      },
-      behaviors: [
-        Polymer.IronResizableBehavior
-      ],
-      observers: [
-        '_populate(currentYear, currentMonth, minDate, maxDate, locale)',
-      ],
-      listeners: {
-        'iron-resize': '_resizeHandler',
-        'swiped': '_onSwipe'
-      },
-      ready: function() {
+          /**
+           * The currently selected year
+           */
+          currentYear: {
+            type: Number
+          },
+          _contentClass: String,
+          _months: Array,
+          _firstDayOfWeek: Number
+        };
+      }
+      static get observers() {
+        return [
+          '_populate(currentYear, currentMonth, minDate, maxDate, locale)'
+        ]
+      }
+      ready() {
+        super.ready();
+        this.addEventListener('iron-resize', this._resizeHandler.bind(this));
```

Comment:
For what i read there, as the listener are on the component there is no memory leak, it would be the case if the listener are set on the window

Reply:
This is keeping an event listener permanently registered, which means that even if it's detached the logic can still be triggered if those events are triggered.  I avoid that in general across the board, however in this case technically it's not an issue for us since those events won't be triggered natively, they are driven by the element being in the DOM, you're good on this one.